### PR TITLE
Update build and run on dirty files

### PR DESCRIPTION
**Pull request by dfithian**
Update LTS, add stack.yaml.lock, and update to allow the --all-dirty flag which will reformat all (git) dirty files. Not sure how SimSpace deals with stack.yaml.lock, so can remove that if desired.

I tested this by opening a file in `vim`, changing import order, and running `simformat --all-dirty`. Got the following output as an example:

```bash
➜  range git:(ps-6667-lookup-old-external-entities) ✗ simformat --all-dirty               
Reformatting hardhat-server/src/Hardhat/Api.hs
Reformatting hardhat-server/src/Hardhat/Dispatcher/SpecificationYaml.hs
```

### CODE REVIEW

**Comment by rickowens** [comment]
I am interested to compare this approach with just wrapping the existing simformat in a bash script. 

On the one hand, philosophically, increasing the complexity and the dependencies of a tool generally cause its long-term maintenance (and therefore usefulness) to suffer.

On the other hand... meh that might be making a mountain out of a mole hill, and we've already started down the road of adding options that alter the behavior of how `simformat` selects its target and outputs.

At least we haven't started adding a bunch of options about how it actually does formatting, and I hope we don't!

Ok, end rambling. This looks ok in general. I'll approve it once I give it a little bit closer study.

File: package.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: simformat
-version: 0.0.0.1
+version: 0.0.0.2
```
Comment (rickowens) [inline package.yaml]: This is an upgrade to the (command-line) interface, so strictly speaking I guess it should be a minor version bump?

File: stack.yaml.lock [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# This file was autogenerated by Stack.
```
Comment (rickowens) [inline stack.yaml.lock]: Culture seems to be that we generally `.gitignore` the `stack.yaml.lock` files.

**Comment by dfithian** [comment]
I definitely thought about wrapping it in a bash script but I decided against it because `simformat ...` is already a one-liner, and I assumed it would be more usable if people who were already using it didn't have to write an alias or reference a bash script somewhere.

**Comment by dfithian** [comment]
The bash scripting alternative would be to do something like:
```bash
git status --porcelain | grep ^\\.\\*\\.hs$ | sed s/^...// | xargs simformat --in-place
```
and then either alias this as something like `simformat-dirty` or stick it in a bash script, but it has to get into `$PATH` somehow.

File: stack.yaml.lock (hunk 2 above)
Comment (gelisam) [reply stack.yaml.lock]: it's not culture, it's experience ;)

We used to track the lock file in git, but this lead to issues, so we stopped tracking it. See [this thread](https://simspace.slack.com/archives/C5SJCF40Y/p1575325119325000?thread_ts=1575318731.316300&cid=C5SJCF40Y) if you're curious about the details.

**Pull request merged**